kasa/protocol.py

```diff
@@ -36,6 +36,14 @@ class TPLinkSmartHomeProtocol:
     def __init__(self, host: str) -> None:
         """Create a protocol object."""
         self.host = host
+        self.port = TPLinkSmartHomeProtocol.DEFAULT_PORT
+        hSplit = host.split(":")
```

Comment:
So, I'm not personally against making this configurable, but the proper way would be to introduce a keyword only `port` argument defaulting to the `DEFAULT_PORT`, so something like:
```
def __init__(self, host: str, *, port = TPLinkSmartHomeProtocol.DEFAULT_PORT):
```

The unit tests would also need to be updated to test for this behavior.

Reply:
Thank you for your comment. If this would be a generic library, I would agree with you. I am always against passing cryptic parameters with some kind of separators.

However, in this case, passing the port number through the colon notation is common knowledge. This is how everybody does it in the browser.

Also, with this approach, we would minimize the scope of impact and the change will automatically enable all existing integrations (in my case I am looking forward to using it in Home Assistant).

Most people won't need a custom port, so having a separate parameter might confuse people, about what to put there even if there is an existing default port.

Reply:
I'd second @rytilahti suggestion. While it would enable this behavior in HASS automatically, which we all use, I believe this host.split(":"), should actually go into the HASS integration, and not in python-kasa. In any event, to use the new feature, one of us would have to open a PR with HASS anyways to bump the python-kasa version, so this minor change can go with this PR as well.

Reply:
The above makes sense, so I modified my PR to pass port separately.